sdk/platform/http_client/curl/src/curl_client.cpp

```diff
@@ -32,5 +33,123 @@ Response CurlClient::send()
     }
   }
 
-  return Response(200, "OK\n");
+  return move(this->m_response);
+}
+
+CURLcode CurlClient::Perform()
+{
+  m_firstHeader = true;
+
+  auto settingUp = SetUrl();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetHeaders();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetWriteResponse();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  return curl_easy_perform(m_pCurl);
+}
+
+static std::unique_ptr<Response> ParseAndSetFirstHeader(std::string const& header)
+{
+  // set response code, http version and reason phrase (i.e. HTTP/1.1 200 OK)
+  auto start = header.begin() + 5; // HTTP = 4, / = 1, moving to 5th place for version
+  auto end = std::find(start, header.end(), '.');
+  auto majorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of minor version
+  end = std::find(start, header.end(), ' ');
+  auto minorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of status code
+  end = std::find(start, header.end(), ' ');
+  auto statusCode = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of reason phrase
+  auto reasonPhrase = std::string(start, header.end() - 2); // remove \r and \n from the end
```

Comment:
This needs to be protected in case `start > header.end()-2` (That is, there is no newline at the end of the header)